### ISSUE TITLE
Could not build Citybound with current nightly

### ISSUE THREAD

**Issue by senden9** [citybound/citybound#235]
First: I know that `CONTRIBUTING.md` says that Citybound should be build with Rust `nightly-2017-10-19` so close this issue if you think it is irrelevant. 

I could not build Citybound  with the current nightly `1.24.0-nightly (560a5da9f 2017-11-27)`. 
```
$ git rev-parse HEAD
288e156851e6ba8020993f85cb012ccd73c1f710
```
```
$ rustup override list
/home/username/tmp/citybound             	nightly-x86_64-unknown-linux-gnu
```
```
$ cargo run --release
   Compiling clippy_lints v0.0.166
   Compiling primal v0.2.3
   Compiling ncollide_math v0.5.0
   Compiling libloading v0.3.4
error: :vis fragment specifier is experimental and subject to change (see issue #41022)
 --> <declare_lint macros>:1:3
  |
1 | ( $ vis : vis $ NAME : ident , $ Level : ident , $ desc : expr ) => (
  |   ^^^^^^^^^^^
  |
  = help: add #![feature(macro_vis_matcher)] to the crate attributes to enable

error: aborting due to previous error

error: Could not compile `clippy_lints`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```
The issue mentioned in the error is rust-lang/rust#41022.

Edit: Same error on the `nov-2017-prototype` (d1a18dddc039e9eea8253a85336410f7f19a44d8) branch.

**Comment by kingoflolz** [comment]
This is because Clippy is closely coupled to the compiler, so if the compiler is updated, so does Clippy.

There is a PR to update to the latest nightly, with the necessary dependency changes, see #217

**Comment by senden9** [comment]
OK. Then, I think, that issue can be closed after the merge of #217.
